modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/Processing.scala

```diff
@@ -97,46 +105,41 @@ object Processing {
       .through(parseBytes(env, badProcessor))
       .through(handleParseFailures(env))
       .through(BatchUp.noTimeout(env.inMemBatchBytes))
-      .through(sinkParsedBatch(env, badProcessor, ref))
+      .through(transformBatch(env, badProcessor, ref))
+      .through(sinkTransformedBatch(env, ref))
```

Comment:
Curious: any specific reason why `finalizeWindow` is not here as another `Pipe` but it's `append`ed later instead?

Reply:
In my mind, `processBatches` is "all the stuff we do for every single batch".  And `finalizeWindow` stands apart because it is something we do once per window.

Syntactically you are right I could add it here like....

```scala
  ...
  .through(transformBatch(env, badProcessor, ref))
  .through(sinkTransformedBatch(env, ref))
  .through(stream => stream.append(finalizeWindow(env, stateRef)))
```

I liked the separation.  But I'd also change it if you like the other way.